components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppLauncherActivity.kt

```diff
@@ -44,11 +51,34 @@ class WebAppLauncherActivity : AppCompatActivity() {
         }
     }
 
+    override fun onResume() {
+        super.onResume()
+
+        val backgroundEndTime = SystemClock.elapsedRealtimeNanos()
+        emitBackgroundTimingFact(backgroundEndTime - backgroundStartTime)
+
+        setBackgroundStartTime(0L) // reset
+    }
+
+    override fun onPause() {
+        super.onPause()
+        setBackgroundStartTime(SystemClock.elapsedRealtimeNanos())
+    }
+
     override fun onDestroy() {
         super.onDestroy()
         scope.cancel()
     }
 
+    @VisibleForTesting(otherwise = PRIVATE)
+    internal fun setBackgroundStartTime(timeNs: Long) {
+        backgroundStartTime = timeNs
+        val bundle = Bundle()
+        bundle.putLong("backgroundStartTime", backgroundStartTime)
+
+        onSaveInstanceState(bundle)
```

Comment:
You do not need to invoke this directly. Android will invoke it before it destroys the activity. So instead of calling it, override it and save the values you want to keep into the bundle that Android passes to you. It will pass them back to you in `onCreate()` if a recreate has happened.

```Kotlin
override fun onSaveInstanceState(bundle: Bundle) {
    bundle.putLong("backgroundStartTime", ....)
}
```

Reply:
Ohhhh!!! That makes more sense 😅